meilisearch/task.py

```diff
@@ -0,0 +1,116 @@
+from time import sleep
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+from meilisearch._httprequests import HttpRequests
+from meilisearch.config import Config
+from meilisearch.errors import MeiliSearchTimeoutError
+
+class Task:
+    """
+    Tasks routes wrapper.
+
+    Task class gives access to all Task routes and child routes via an Index (inherited).
+    https://docs.meilisearch.com/reference/api/task.html
+    """
+
+    def __init__(
+        self,
+        config: Config,
+        uid: Optional[str] = None,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        config:
+            Config object containing permission and location of MeiliSearch.
+        """
+        self.config = config
+        self.http = HttpRequests(config)
+        self.uid = uid
+
+    def get_tasks(self) -> Dict[str, List[Dict[str, Any]]]:
+        """Get all tasks.
+
+        Returns
+        -------
+        task:
+            Dictionary containing a list of all enqueued, processing, succeeded or failed tasks.
+
+        Raises
+        ------
+        MeiliSearchApiError
+            An error containing details about why MeiliSearch can't process your request. MeiliSearch error codes are described here: https://docs.meilisearch.com/errors/#meilisearch-errors
+        """
+        if self.uid is None:
+            return self.http.get(
+                f'{self.config.paths.task}'
+            )
+        return self.http.get(
+            f'{self.config.paths.index}/{self.uid}/{self.config.paths.task}'
```

Comment:
This isn't needed is it? According to the spec it looks like all tasks go to a single url so there isn't a need for a separate index url. I think all that is needed is:

```py
return self.http.get(f'{self.config.paths.task}'
```

Reply:
Sorry, I see you can do it both ways so the index id is needed